src/TorchSharp/Data/Dataset.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Collections.Generic;
+
+using static TorchSharp.torch;
+
+namespace TorchSharp.Data
+{
+    public interface Dataset: IDisposable
```

Comment:
`IDataset` maybe?

Reply:
I did like that because pytorch does.

Reply:
IDataset is right for a .NET interface. 

The 'Dataset' class is abstract in PyTorch, so please change to an abstract class, or change the name.